src/test/java/com/linkedin/kmf/services/ConsumeServiceTest.java

```diff
@@ -0,0 +1,186 @@
+
+
+/**
+ * Unit Testing for the Consume Service Class.
+ * Also tests for Kafka Monitor Consumer offset commits.
+ */
+public class ConsumeServiceTest {
+  private static final String TOPIC = "kafka-monitor-topic-testing";
+  private static final Logger LOG = LoggerFactory.getLogger(ConsumeServiceTest.class);
+  private CommitAvailabilityMetrics _commitAvailabilityMetrics;
+  private static final String TAGS_NAME = "name";
+  private static final String METRIC_GROUP_NAME = "commit-availability-service";
+  private static Boolean testCommittedSuccessfully = false;
+
+  @Test
+  public void lifecycleTest() throws Exception {
+    ConsumeService consumeService = consumeService();
+
+    /* Nothing should be started */
+    Assert.assertFalse(consumeService.isRunning());
+    Assert.assertNotNull(consumeService.getServiceName());
+
+    /* Should accept but ignore start because start has not been called */
+    consumeService.stop();
+    Assert.assertFalse(consumeService.isRunning());
+
+    /* Should start */
+    consumeService.start();
+    Assert.assertTrue(consumeService.isRunning());
+
+    /* Should allow start to be called more than once */
+    consumeService.stop();
+    consumeService.stop();
+    Assert.assertFalse(consumeService.isRunning());
+
+    /* Should be allowed to shutdown more than once. */
+    consumeService.awaitShutdown();
+    consumeService.awaitShutdown();
+    Assert.assertFalse(consumeService.isRunning());
+  }
+
+  @Test
+  public void commitAvailabilityTest() throws Exception {
+    ConsumeService consumeService = consumeService();
+
+    MetricConfig metricConfig = new MetricConfig().samples(60).timeWindow(1000, TimeUnit.MILLISECONDS);
+    List<MetricsReporter> reporters = new ArrayList<>();
+    reporters.add(new JmxReporter(Service.JMX_PREFIX));
+    Metrics metrics = new Metrics(metricConfig, reporters, new SystemTime());
+    String name = "name";
+    Map<String, String> tags = new HashMap<>();
+    tags.put(TAGS_NAME, name);
+
+    _commitAvailabilityMetrics = new CommitAvailabilityMetrics(metrics, tags);
+    Assert.assertNotNull(_commitAvailabilityMetrics._offsetsCommitted.name());
+    Assert.assertNotNull(metrics.metrics().get(metrics.metricName("offsets-committed-total", METRIC_GROUP_NAME, tags)).metricValue());
+    Assert.assertEquals(metrics.metrics().get(metrics.metricName("offsets-committed-total", METRIC_GROUP_NAME, tags)).metricValue(), 0.0);
+
+    /* Should start */
+    consumeService.start();
+    Assert.assertTrue(consumeService.isRunning());
+
+    long threadStartDelay = 2000;
+
+    /* Thread.sleep safe to do here instead of ScheduledExecutorService
+    *  We want to sleep current thread so that consumeService can start running for enough seconds. */
+    Thread.sleep(threadStartDelay);
+    Assert.assertNotNull(metrics.metrics().get(metrics.metricName("offsets-committed-total", METRIC_GROUP_NAME, tags)).metricValue());
+    Assert.assertTrue(testCommittedSuccessfully);
```

Comment:
This doesn't test that the ConsumeService processed the message correctly and recorded a count.

Reply:
Updated.